### ISSUE TITLE
HTTPResponse.read1() related test failures

### ISSUE THREAD

**Issue by abebeos** [urllib3/urllib3#3249]
### Subject

Test failures in this PR:

- https://github.com/urllib3/urllib3/pull/3248

### Environment

See CI

### Steps to Reproduce

See CI

### Expected Behavior

Tests should pass

### Actual Behavior

https://github.com/urllib3/urllib3/actions/runs/7352783311/job/20017967313?pr=3248#step:7:572

Test Code:

https://github.com/urllib3/urllib3/blob/dcf69a3d8699eff80512d42376080df6a0104b53/test/test_response.py#L451-L488

`read1` Code:

https://github.com/urllib3/urllib3/blob/dcf69a3d8699eff80512d42376080df6a0104b53/src/urllib3/response.py#L951-L1000

### Alternative

~Should this be not fixable, then add/update the documentation a bit to avoid misunderstandings.~

**Comment by abebeos** [comment]
(something tells me that I've possibly 'lost it' here. But in the end, this just leads to better documentation)

**Comment by abebeos** [comment]
@sethmlarson , I've now the confidence that those test-failures are real (e.g. `read1` code is broken. This is highly inconsistent:

```sh
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[foofoofoofoofoo] PASSED [ 26%]
test/test_response.py::TestResponse::test_decode_zstd_incomplete_read1[123456789012345] FAILED [ 26%]
```

Should be worth a bounty?

(Me currently 'warm', thus continuing. I've a clear plan in mind how to fix and at the same time make things a bit more clear)

**Comment by abebeos** [comment]
this seems to work locally, **all tests pass** (windows machine, some tests not executed, but most possibly not relevant for read/read1)

**Comment by abebeos** [comment]
Fixed with 49ac228e07ca95c1608e89d31ab08fae71c561f6

**Comment by abebeos** [comment]
Superseded by #3250
